test/new-e2e/tests/apm/docker_test.go

```diff
@@ -0,0 +1,159 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package apm
+
+import (
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awsdocker "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/docker"
+	"github.com/DataDog/test-infra-definitions/components/datadog/dockeragentparams"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/stretchr/testify/assert"
+)
+
+type DockerFakeintakeSuite struct {
+	e2e.BaseSuite[environments.DockerHost]
+}
+
+func TestDockerFakeintakeSuite(t *testing.T) {
+	devModeEnv, _ := os.LookupEnv("E2E_DEVMODE")
+	options := []e2e.SuiteOption{
+		e2e.WithProvisioner(awsdocker.Provisioner(awsdocker.WithAgentOptions(
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Note: docker support is currently broken as VMs cannot access the images that we publish to the `agent-qa` ECR. @KevinFairise2 is working on it, should be done by end of next week.

Reply:
Do you think it's a blocker for merging the PR?

Reply:
Unfortunately yes :(. You could add it and allow to fail it or add it and skip it with the message ("blocked by APL-2786")

Reply:
It runs fine locally, it will fail on main as we use the agent image published to the `agent-qa` ECR

Reply:
Will do, thanks!

Reply:
Will add the suggestion where this should happen